src/wallet/wallet.cpp

```diff
@@ -778,8 +778,16 @@ void CWalletTx::GetAmounts(list<COutputEntry>& listReceived, list<COutputEntry>&
 
     strSentAccount = strFromAccount;
 
+    // This is the same as nDebit > 0, i.e. we sent the transaction.
     bool fIsFromMe = IsFromMe();
 
+    // This will be true if this is a self-transaction.
+    bool fIsAllToMe = true;
+    for (auto const& txout : vout)
+    {
+        fIsAllToMe = fIsAllToMe && (pwallet->IsMine(txout) != ISMINE_NO);
```

Comment:
We can break early after determining that `fIsAllToMe == false` to avoid redundant scans over big transactions.

Reply:
Yep. Good catch. I should fix the same loop on the UI side! :)